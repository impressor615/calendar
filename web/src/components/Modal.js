import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Button from 'components/Button';

export const ModalHeader = ({ children, onToggle }) => (
  <div className="modal-header">
    <div>{children}</div>
    <Button onClick={onToggle}>&times;</Button>
  </div>
);

ModalHeader.propTypes = {
  children: PropTypes.node.isRequired,
  onToggle: PropTypes.func.isRequired,
};

export const ModalBody = ({ children }) => (
  <div className="modal-body">
    {children}
  </div>
);

ModalBody.propTypes = {
  children: PropTypes.node.isRequired,
};

export const ModalFooter = ({ children }) => (
  <div className="modal-footer">
    { children }
  </div>
);

ModalFooter.propTypes = {
  children: PropTypes.node.isRequired,
};

export const ModalDialog = ({ children, ...rest }) => (
  <div className={classnames('modal-dialog', rest.className)}>
    { children }
  </div>
);

ModalDialog.propTypes = {
  children: PropTypes.node.isRequired,
};

const Modal = ({ isOpen, children, backdrop, onToggle, ...rest }) => (
  isOpen ? (
    ReactDOM.createPortal(
      <div
        {...rest}
        className={classnames('modal', rest.className)}
      >
        {children}
        { backdrop ? <div className="modal-backdrop" onClick={onToggle} /> : null }
      </div>,
      document.getElementById('modal-root'),
    )
  ) : null
);

Modal.defaultProps = {
  backdrop: true,
  onToggle: () => {},
};

Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onToggle: PropTypes.func,
  backdrop: PropTypes.bool,
};

export default Modal;
