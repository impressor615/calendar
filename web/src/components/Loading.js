import React from 'react';
import PropTypes from 'prop-types';

import Modal, { ModalDialog } from 'components/Modal';



const Loading = ({ isOpen, children }) => (
  <Modal isOpen={isOpen}>
    <ModalDialog className="center">
      {children}
    </ModalDialog>
  </Modal>
);

Loading.defaultProps = {
  children: 'Loading...',
};

Loading.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  children: PropTypes.node,
};

export default Loading;
