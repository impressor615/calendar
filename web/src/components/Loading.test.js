import React from 'react';

import Loading from 'components/Loading';
import Modal, { ModalDialog } from 'components/Modal';


describe('<Loading />', () => {
  const modalRoot = document.createElement('div');
  modalRoot.setAttribute('id', 'modal-root');
  const body = document.querySelector('body');
  body.appendChild(modalRoot);

  let wrapper;
  const props = {
    isOpen: true,
    children: 'Loading...' 
  };

  afterEach(() => {
    wrapper.unmount();
  })

  it('should be rendered properly', () => {
    wrapper = mount(<Loading {...props} />);
    expect(wrapper.find(Modal)).toHaveLength(1);
    expect(wrapper.find(ModalDialog)).toHaveLength(1);
    expect(wrapper.text()).toEqual(props.children);
  })

  it('shoud be matched with the snapshot', () => {
    wrapper = mount(<Loading {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
});

