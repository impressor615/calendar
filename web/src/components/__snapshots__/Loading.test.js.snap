// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> shoud be matched with the snapshot 1`] = `
<Loading
  isOpen={true}
>
  <Modal
    backdrop={true}
    isOpen={true}
    onToggle={[Function]}
  >
    <Portal
      containerInfo={
        <div
          id="modal-root"
        >
          <div
            class="modal"
          >
            <div
              class="modal-dialog center"
            >
              Loading...
            </div>
            <div
              class="modal-backdrop"
            />
          </div>
        </div>
      }
    >
      <div
        className="modal"
      >
        <ModalDialog
          className="center"
        >
          <div
            className="modal-dialog center"
          >
            Loading...
          </div>
        </ModalDialog>
        <div
          className="modal-backdrop"
          onClick={[Function]}
        />
      </div>
    </Portal>
  </Modal>
</Loading>
`;
